trieste/models/model_interfaces.py

```diff
@@ -24,24 +24,14 @@
 from ..type import ObserverEvaluations, QueryPoints, TensorType
 
 
-class ModelInterface(ABC):
-    """ A trainable probabilistic model. """
-
-    @abstractmethod
-    def update(self, dataset: Dataset) -> None:
-        """
-        Update the model given the specified ``dataset``. Does not train the model.
-
-        :param dataset: The data with which to update the model.
-        """
-        raise NotImplementedError
+class ProbabilisticModel(ABC):
```

Comment:
this class is the main change: an untrainable probabilistic model, which is a superclass of the old `ModelInterface` (now called `TrainableProbabilisticModel`)